### PR TITLE
Address Carrick's review.

### LINKED ISSUE #44: Why use both certificates and a PSK on the same connection?
(This question is copied from my email.)

The draft states: "To perform rare system specific operations that require a higher security level, the central server can request resource-intensive client authentication with the usage of a certificate after successfully establishing the connection with a PSK."

If you're going to authenticate with a cert, why bother preceding that 
with an authentication with a PSK?

### PR DESCRIPTION
**Pull request by jhoyla**
Closes #44, #45, and #47.

### CODE REVIEW

**Comment by jhoyla** [comment]
@cbartle891 does this address your (other) comments?

**Comment by cbartle891** [comment]
I think this adds really helpful context to what would otherwise be confusing on its own, thanks.

File: draft-ietf-tls-external-psk-guidance.md [hunk 1]
```diff
@@ -233,10 +233,12 @@ connections with early data.
...
-- Internet of Things (IoT) and devices with limited computational capabilities. {{?RFC7925}}
-defines TLS and DTLS profiles for resource-constrained devices and suggests the use of PSK
-ciphersuites for compliant devices. The Open Mobile Alliance Lightweight Machine to
-Machine Technical Specification {{LwM2M}} states that LwM2M servers MUST support the PSK mode of DTLS.
+- Internet of Things (IoT) and devices with limited computational capabilities. {{?RFC7925}} defines TLS and DTLS
+  profiles for resource-constrained devices and suggests the use of PSK ciphersuites for compliant devices. The Open
+Mobile Alliance Lightweight Machine to Machine Technical Specification {{LwM2M}} states that LwM2M servers MUST support
+the PSK mode of DTLS. If all devices may communicate with each other then to achieve full security n squared keys need
+to be provisioned. If this would be overly burdensome it is possible to achieve some level of security by distrubuting n
+keys, and operating a central server.
```
Comment (chris-wood) [inline draft-ietf-tls-external-psk-guidance.md]: "full security" seems too vague. I'd recommend reverting this change, as I don't think it adds much and isn't consistent with the other items in this list. (This bullet item is identifying IoT as a use case, not explaining all the ways in which IoT systems may use PSKs, and why those certain ways are good or bad or whatever.)

File: draft-ietf-tls-external-psk-guidance.md [hunk 2]
```diff
@@ -255,11 +257,14 @@ distributed out-of-band to the group participants. Members can then establish pe
...
-operations that require a higher security level, the central server can request resource-intensive client
-authentication with the usage of a certificate after successfully establishing the connection with a PSK.
+- Internet of Things (IoT). In this use-case, resource-constrained IoT devices act as TLS clients and share the same
+  PSK. The devices use this PSK for quickly establishing connections with a central server. Such a scheme ensures that
+the client IoT devices are legitimate members of the system. To perform rare system specific operations that require a
+higher security level, the central server can request resource-intensive client authentication with the usage of a
+certificate after successfully establishing the connection with a PSK. The use of a single PSK for multiple devices in
+this way is not recommended, but may be seen in cases where there is a large turnover in the number of devices. The use
+of connection upgrades may be employed where it is not known in advance whether the higher level of authentication is
+necessary, or to bind a particular action to a particular device.
```
Comment (chris-wood) [inline draft-ietf-tls-external-psk-guidance.md]:
```suggestion
- Internet of Things (IoT) and devices with limited computational capabilities. Many PSK provisioning examples are possible in this use-case. For example, in a given setting, IoT devices may all share the same PSK and use it to communicate with a central server (one key for n devices), have their own key for communicating with a central server (n keys for n devices), or have pairwise keys for communicating with each other (n^2 keys for n devices). 

The exact provisioning process depends on the system requirements and threat model. Generally, use of 
a single PSK shared between more than one node is not recommended, even if other accommodations 
are made, such as client certificate authentication after PSK-based connection establishment. 
See {{recommendations}}.
```

File: draft-ietf-tls-external-psk-guidance.md [hunk 3]
```diff
@@ -255,11 +257,14 @@ distributed out-of-band to the group participants. Members can then establish pe
 with each other using the external PSK. It is important to note that any node of the group can behave
```
Comment (chris-wood) [inline draft-ietf-tls-external-psk-guidance.md]: Unrelated to your change, but I'd also suggest rewriting this item:

```suggestion
- Group chats. In this use-case, group participants may be provisioned an external PSK out-of-band for establishing authenticated connections with other members of the group.
```

**Review by chris-wood — CHANGES_REQUESTED** [review]

**Pull request merged**